src/mlpack/core/data/normalize_labels_impl.hpp

```diff
@@ -39,32 +40,32 @@ void NormalizeLabels(const RowType& labelsIn,
   // we'll resize it back down to its actual size.
   mapping.set_size(labelsIn.n_elem);
   labels.set_size(labelsIn.n_elem);
+  // Map for mapping labelIn to their label
+  std::unordered_map<eT, size_t> labelMap;
   size_t curLabel = 0;
   for (size_t i = 0; i < labelsIn.n_elem; ++i)
   {
-    bool found = false;
-    for (size_t j = 0; j < curLabel; ++j)
+    // If labelsIn[i] is already in the map, use the existing label.
+    if (labelMap.count(labelsIn[i]) > 0)
     {
-      // Is the label already in the list of labels we have seen?
-      if (labelsIn[i] == mapping[j])
-      {
-        labels[i] = j;
-        found = true;
-        break;
-      }
+      labels[i] = labelMap[labelsIn[i]] - 1;
```

Comment:
I don't follow why the `- 1` and `+ 1` throughout; you could adapt the code to just do `labels[i] = labelMap[labelsIn[i]]` and this would be cleaner.  I'd like it if you could do that. :+1:

Reply:
Ahaa, I completely forgot about it to change. 

I guess if you take a look at my previous version of code it was `if (labelMap[labelsIn[i]]  != 0) ` instead of `(labelMap.count(labelsIn[i]) > 0) ` and thus i was forced to not insert 0, becasue i had to check the presence of an element .